--- autofocus.js	2016-05-09 07:15:14.333132210 -0400
+++ autofocus.js.new	2016-05-09 07:09:10.016132210 -0400
@@ -8,8 +8,10 @@
         if(attr.autoFocus == "" || attr.autoFocus == "proactiveFocus") {
           angular.element(scope['form_focus_' + ctrl.$name]).focus();
           if (!angular.element(scope['form_focus_' + ctrl.$name]).is(":focus")) {
+            setMiqQE('autofocus', 'waiting');
             $timeout(function () {
               angular.element(scope['form_focus_' + ctrl.$name]).focus();
+              setMiqQE('autofocus', 'complete');
             }, 1000);
           }
         }
@@ -17,8 +19,10 @@

       scope.$on('reactiveFocus', function(e) {;
         if (!angular.element(scope['form_focus_' + ctrl.$name]).is(":focus")) {
+          setMiqQE('autofocus', 'waiting');
           $timeout(function(){
             angular.element(scope['form_focus_' + ctrl.$name]).focus();
+            setMiqQE('autofocus', 'complete');
           }, 0);
         };
       });
