diff --git a/vmdb/public/javascripts/application.js b/vmdb/public/javascripts/application.js
--- a/vmdb/public/javascripts/application.js
+++ b/vmdb/public/javascripts/application.js
@@ -568,13 +568,6 @@
 //	wrapFish();
 //});
 
-// Register prototype Ajax Responders
-Ajax.Responders.register({
-	onException: function(parm1, parm2) {
-		Ajax.activeRequestCount-- // decrement request count when ajax throws an exception (workaround)
-	}
-});
-
 // Load XML/SWF charts data (non-IE)
 // This method is called by the XML/SWF charts when a chart is loaded into the DOM
 var miq_chart_data;
